authn.js

```diff
@@ -210,25 +210,33 @@ function setup(app) {
   });
 
   /**
-   * Returns an array of names of Nextstrain groups that are visible to a
-   * given *user*.
+   * Returns an array of Nextstrain groups that are visible to a
+   * given *user* (or a non-logged in user). The order of groups returned
+   * matches the `sources` array in `sources.js`.
    *
    * FIX: Contains a hard-coded assumption that all Nextstrain groups match
    * their corresponding group name exactly.
    * See <https://github.com/nextstrain/nextstrain.org/issues/76> for more
    * context and to track this issue.
    *
-   * @param {Object} user
-   * @returns {Array}
+   * @param {Object | undefined} user. `undefined` represents a non-logged-in user
+   * @returns {Array} Each element is an object with keys `name` -> {str} (group name) and
+   *                  `private` -> {bool} (group is private)
    */
-  const visibleGroups = (user) => Array.from(sources.values())
-    .filter(source => source.visibleToUser(user))
-    .map(source => source._name)
-    .filter(source => user.groups.includes(source));
+  const visibleGroups = (user) => Array.from(sources)
+      .filter(([, source]) => source.isGroup())
+      .filter(([, source]) =>
+        source.isPrivate() ?
+          source.visibleToUser(user) :
+          true /* allow all public sources regardless of user data */
```

Comment:
The use of `isPrivate()` here subverts the intent that `visibleToUser(…)` is the sole authorizer of a (anonymous or logged in) user's ability to see a Source. As noted above, the `isPrivate()` method is unnecessary, and I think it should be removed.

Reply:
👍 resolved